python/pyspark/sql/dataframe.py

```diff
@@ -470,7 +470,7 @@ def sampleBy(self, col, fractions, seed=None):
         +---+-----+
 
         """
-        if not isinstance(col, str):
+        if not isinstance(col, basestring):
```

Comment:
Thank you for answering. I understood why `isinstance(col, basestring)` is not needed here.

Although column name is basically stored as `str`, it is stored as `unicode` in a certain case.
See [SPARK-15244](https://issues.apache.org/jira/browse/SPARK-15244) for details.